zebra-consensus/src/block/check.rs

```diff
@@ -246,9 +260,19 @@ pub fn miner_fees_are_valid(
     // https://zips.z.cash/protocol/protocol.pdf#txnconsensus
     let left = (transparent_value_balance - sapling_value_balance - orchard_value_balance)
         .map_err(|_| SubsidyError::SumOverflow)?;
-    let right = (block_subsidy + block_miner_fees).map_err(|_| SubsidyError::SumOverflow)?;
+    let right = (block_subsidy + block_miner_fees - expected_deferred_amount)
+        .map_err(|_| SubsidyError::SumOverflow)?;
+
+    // TODO: Add link to exact coinbase balance ZIP
+    let should_allow_unclaimed_subsidy =
+        NetworkUpgrade::current(network, height) <= NetworkUpgrade::Nu5;
+    let is_invalid_miner_fee = if should_allow_unclaimed_subsidy {
+        left > right
+    } else {
+        left != right
+    };
 
-    if left > right {
+    if is_invalid_miner_fee {
         Err(SubsidyError::InvalidMinerFees)?;
     }
```

Comment:
The current title ("change(consensus): Add lockbox funding stream") of the PR does not reflect that it includes this change. Also, I would suggest making it a separate PR for ease of review. (The other changes aren't dependent on it, I think.)

Reply:
Reverted changes related to the coinbase-should-balance-exactly ZIP in https://github.com/ZcashFoundation/zebra/pull/8694/commits/4627c0854dc40d65cf7a33ee16645a7c8834ef48, I'll open another PR for it.